engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -267,21 +269,53 @@ namespace dmGameSystem
         }
     }
 
+    static inline MaterialResource* GetMaterialResource(const ModelComponent* component, const ModelResource* resource, uint32_t index) {
+        // TODO: Add support for setting material on different indices
+        return component->m_Material ? component->m_Material : resource->m_Materials[index].m_Material;;
+    }
+
     static inline dmRender::HMaterial GetMaterial(const ModelComponent* component, const ModelResource* resource, uint32_t index) {
-        if (component->m_Material)
-            return component->m_Material; // TODO: Add support for setting material on different indices
-        return resource->m_Materials[index];
+        return GetMaterialResource(component, resource, index)->m_Material;
+    }
+
+    TextureResource* GetTextureResource(const ModelComponent* component, uint32_t material_index, uint32_t texture_unit)
```

Comment:
Yeah maybe we might as well generalise this now since it's the same code as for meshes